### PR TITLE
Initial Prow deployment

### PR DESCRIPTION
**Pull request by icelynjennings**
Adds Prow config, plugins, manifests, verification and branding

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      path_strategy: "explicit"
+    gcs_credentials_secret: "prow-gcs"
+
+sinker:
+  resync_period: 1h
+  max_prowjob_age: 48h
+  max_pod_age: 1h
+
+deck:
+  branding:
```
Comment (icelynjennings) [inline config/config.yaml]: This is in-built? Shocked.

File: prow/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: prow
+  namespace: default
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/ingress.global-static-ip-name: "prow-nginx"
```
Comment (icelynjennings) [inline prow/ingress.yaml]: rm or leave?

**Comment by icelynjennings** [comment]
Other than that LGTM

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+ECHO := echo
+
+.PHONY: generate-config
+generate-config:
```
Comment (JoelSpeed) [inline Makefile]: @gargath You wanted us to rename this target IIRC?

File: prow/ingress.yaml (hunk 2 above)
Comment (JoelSpeed) [reply prow/ingress.yaml]: Yep, that needs removing

**Review by JoelSpeed — COMMENTED** [review]
I'm generally happy with this now, @icelynjennings Could you remove that IP annotation from the ingress and then let's get @mthssdrbrg and @gargath to also do a quick review

**Comment by icelynjennings** [comment]
Rebased

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	@ echo # Produce a new line at the end of each target to help readability
+
+.PHONY: verify-config
+verify-config: checkconfig
+	@ $(ECHO) "\033[36mVerifying Config\033[0m"
+	${GOPATH}/bin/checkconfig --config-path=config/config.yaml --job-config-path=config/jobs --plugin-config=config/plugins.yaml
+	@ echo # Spacer between output
+	make generate-config
+	@ $(ECHO) "\033[36mVerifying Git Status\033[0m"
+	@ if [ "$$(git status -s)" != "" ]; then git diff --color; echo "\033[31;1mERROR: Git Diff found. Please run \`make generate-config\` and commit the result.\033[0m"; exit 1; else echo "\033[32mValid config found\033[0m";fi
```
Comment (gargath) [inline Makefile]: If you rename the target, you'll also have to rename it in the output here.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+	@ echo # Produce a new line at the end of each target to help readability
+
+checkconfig:
```
Comment (gargath) [inline Makefile]: Could this be `$(GOPATH)/bin/checkconfig` so that `make` doesn't re-install it if present?

File: config/config.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    logo: "extensions/logo.svg"
+    favicon: "extensions/favicon.ico"
+    header_color: "#300D4F"
+  spyglass:
+    size_limit: 500000000
+    viewers:
+      "started.json|finished.json": ["metadata-viewer"]
+      "build-log.txt": ["build-log-viewer"]
+      "artifacts/junit.*\\.xml": ["junit-viewer"] # Remember to escape your '\' in yaml strings!
```
Comment (gargath) [inline config/config.yaml]: 😆

**Review by wonderhoss — APPROVED** [review]
Looks sensible to me. The suggested chance to the `Makefile` you can take or leave.

File: scripts/make-jobs-config.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#/env/bin/bash
+
+config=""
+for f in $(ls config/jobs/*/*.yaml); do
+  k=${f#"config/jobs/"}
+  k=$(echo $k | sed "s|/|_|")
+  config="$config --from-file=$k=$f"
+done
+kubectl create configmap jobs ${config} -n default --dry-run -o yaml
```
Comment (mthssdrbrg) [inline scripts/make-jobs-config.sh]: Nitpick; use long options in scripts :)

**Review by mthssdrbrg — COMMENTED** [review]
It looks fine.

Personally I'd use `apps/v1` for the deployments rather than `extensions/v1beta1` and not have double quotes around various strings that don't need it, but they're just nitpicks from me and nothing that should block this from being merged :)

**Review by mthssdrbrg — APPROVED** [review]

**Pull request merged**